### PR TITLE
docs: revamp INSTALL.md

### PR DESCRIPTION
**Pull request by Roasbeef**
In this commit, we aim to revamp the `INSTALL.md` file in order to make
it easier to parse, and more explicit. We've added a table of contents
to allow users to skip down to relevant sections and obtain a snapshot
of the contents of the document. Additionally, we've now modified the
headings to express a more clear delineation of the various sections.

### CODE REVIEW

File: docs/INSTALL.md [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+  * [Bitcoind Options](#bitcoind-options)
+  * [Using btcd](#using-btcd)
+    * [Installing btcd](#installing-btcd)
+    * [Starting btcd](#starting-btcd)
+    * [Running lnd using the btcd backend](#running-lnd-using-the-btcd-backend)
+  * [Using Neutrino](#using-neutrino)
+  * [Using bitcoind or litecoind](#using-bitcoind-or-litecoind)
+* [Macaroons](#macaroons)
+* [Network Reachability](#network-reachability)
+* [Simnet vs. Testnet Development](#simnet-vs.-testnet-development)
```
Comment (xsb) [inline docs/INSTALL.md]:
```suggestion
* [Simnet vs. Testnet Development](#simnet-vs-testnet-development)
```

File: docs/INSTALL.md [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+* [Creating an lnd.conf (Optional)](#creating-an-lnd.conf-(optional))
```
Comment (xsb) [inline docs/INSTALL.md]:
```suggestion
* [Creating an lnd.conf (Optional)](#creating-an-lndconf-optional)
```

File: docs/INSTALL.md (hunk 1 above)
Comment (Roasbeef) [reply docs/INSTALL.md]: Fixed!

**Comment by Roasbeef** [comment]
Thanks for the review! I used some random Ruby script off Stack Overflow, and I guess it didn't handle those cases properly lol.

File: docs/INSTALL.md (hunk 2 above)
Comment (Roasbeef) [reply docs/INSTALL.md]: Fixed!

File: docs/INSTALL.md [hunk 3]
```diff
@@ -311,13 +360,34 @@ using the `--macaroonpath` argument.
...
+`--bitcoin.testnet` flag.
+
+Another relevant command line flag for local testing of new `lnd` developments
+is the `--debughtlc` flag. When starting `lnd` with this flag, it'll be able to
+automatically settle a special type of HTLC sent to it. This means that you
+won't need to manually insert invoices in order to test payment connectivity.
+To send this "special" HTLC type, include the `--debugsend` command at the end
+of your `sendpayment` commands.
+
+
```
Comment (cfromknecht) [inline docs/INSTALL.md]: extra new line

**Pull request merged**